models/repo_collaboration.go

```diff
@@ -183,3 +183,33 @@ func (repo *Repository) DeleteCollaboration(uid int64) (err error) {
 
 	return sess.Commit()
 }
+
+func (repo *Repository) getRepoTeams(e Engine) (teams []*Team, err error) {
+	return teams, e.
+		Join("INNER", "team_repo", "team_repo.team_id = team.id").
+		Where("team.org_id = ?", repo.OwnerID).
+		And("team_repo.repo_id=?", repo.ID).
+		OrderBy("CASE WHEN name LIKE '" + ownerTeamName + "' THEN '' ELSE name END").
+		Find(&teams)
+}
+
+// GetRepoTeams gets the list of teams that has access to the repository
+func (repo *Repository) GetRepoTeams() ([]*Team, error) {
+	return repo.getRepoTeams(x)
+}
+
+// ModeI18nKey returns the i18n key to describe collaboration modes
+func (t *Team) ModeI18nKey() string {
```

Comment:
This should be moved into template

Reply:
In what way, just an if-else-if case? Collaboration has the same structure and also a test case for it (which I don't see much point of)

Reply:
Yes, if-elseif-else. Imho i18n key is not related to model in any way, it should live in presentation layer